akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/internal/ClusterShardingImpl.scala

```diff
@@ -91,9 +98,10 @@ import akka.japi.function.{ Function ⇒ JFunction }
   // typeKey.name to messageClassName
   private val regions: ConcurrentHashMap[String, String] = new ConcurrentHashMap
   private val proxies: ConcurrentHashMap[String, String] = new ConcurrentHashMap
+  private val shardCommandActors: ConcurrentHashMap[String, ActorRef[scaladsl.ClusterSharding.ShardCommand]] = new ConcurrentHashMap
 
   override def spawn[A](
-    behavior:           String ⇒ Behavior[A],
+    behavior:           (ActorRef[scaladsl.ClusterSharding.ShardCommand], String) ⇒ Behavior[A],
```

Comment:
Could it be simpler if we had an actual class here instead of a tuple, would allow giving the fields nice explanatory names and docs etc?

Reply:
Ah, ofc. not sure what I was thinking with...